scrapy/linkextractors/htmlparser.py

```diff
@@ -39,7 +40,7 @@ def _extract_links(self, response_text, response_url, response_encoding):
         ret = []
         base_url = urljoin(response_url, self.base_url) if self.base_url else response_url
         for link in links:
-            if isinstance(link.url, unicode):
+            if isinstance(link.url, six.text_type):
```

Comment:
After lengthy discussions we (me and @dangra) decided that urls should be native strings, i.e. bytes in Python 2 and unicode in Python 3.